libafl/src/feedbacks/mod.rs

```diff
@@ -47,77 +47,239 @@ where
+        let b = self.second.is_interesting(input, observers, exit_kind)?;
+        Ok(a || b)
     }
+}
 
+impl<A, B, I> Named for OrFeedback<A, B, I>
+where
+    A: Feedback<I>,
+    B: Feedback<I>,
+    I: Input,
+{
     #[inline]
-    fn append_metadata_all(&mut self, _testcase: &mut Testcase<I>) -> Result<(), Error> {
-        Ok(())
+    fn name(&self) -> &str {
+        //format!("Or({}, {})", self.first.name(), self.second.name())
+        "OrFeedback"
     }
+}
 
-    #[inline]
-    fn discard_metadata_all(&mut self, _input: &I) -> Result<(), Error> {
-        Ok(())
+impl<A, B, I> OrFeedback<A, B, I>
+where
+    A: Feedback<I>,
+    B: Feedback<I>,
+    I: Input,
+{
+    pub fn new(first: A, second: B) -> Self {
+        Self {
+            first,
+            second,
+            phantom: PhantomData,
+        }
     }
 }
 
-impl<Head, Tail, I> FeedbacksTuple<I> for (Head, Tail)
+/// Compose feedbacks with an OR operation
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(bound = "I: serde::de::DeserializeOwned")]
+pub struct NotFeedback<A, I>
+where
+    A: Feedback<I>,
+    I: Input,
+{
+    pub first: A,
+    phantom: PhantomData<I>,
+}
+
+impl<A, I> Feedback<I> for NotFeedback<A, I>
 where
-    Head: Feedback<I>,
-    Tail: FeedbacksTuple<I>,
+    A: Feedback<I>,
     I: Input,
 {
-    fn is_interesting_all<OT>(
+    fn is_interesting<OT>(
         &mut self,
         input: &I,
         observers: &OT,
         exit_kind: &ExitKind,
-    ) -> Result<u32, Error>
+    ) -> Result<bool, Error>
     where
         OT: ObserversTuple,
     {
-        Ok(self.0.is_interesting(input, observers, exit_kind)?
-            + self.1.is_interesting_all(input, observers, exit_kind)?)
+        Ok(!self.first.is_interesting(input, observers, exit_kind)?)
+    }
+}
+
+impl<A, I> Named for NotFeedback<A, I>
+where
+    A: Feedback<I>,
+    I: Input,
+{
+    #[inline]
+    fn name(&self) -> &str {
+        //format!("Not({})", self.first.name())
+        "NotFeedback"
     }
+}
 
-    fn append_metadata_all(&mut self, testcase: &mut Testcase<I>) -> Result<(), Error> {
-        self.0.append_metadata(testcase)?;
-        self.1.append_metadata_all(testcase)
+impl<A, I> NotFeedback<A, I>
+where
+    A: Feedback<I>,
+    I: Input,
+{
+    pub fn new(first: A) -> Self {
+        Self {
+            first,
+            phantom: PhantomData,
+        }
     }
+}
+
+/// Variadic macro to create a chain of AndFeedback
+#[macro_export]
+macro_rules! feedback_and {
+    ( $last:expr ) => { $last };
 
-    fn discard_metadata_all(&mut self, input: &I) -> Result<(), Error> {
-        self.0.discard_metadata(input)?;
-        self.1.discard_metadata_all(input)
+    ( $head:expr, $($tail:expr), +) => {
+        // recursive call
+        $crate::feedbacks::AndFeedback::new($head , feedback_and!($($tail),+))
+    };
+}
+
+/// Variadic macro to create a chain of OrFeedback
+#[macro_export]
+macro_rules! feedback_or {
```

Comment:
Maybe `feedbacks_or`?

(or even `feedbacks_any`?

Reply:
I want to maintain the names consisent, and feedbacks_not does not make sense as it takes just one parameter

Reply:
And in the end feedback_or creates a new OrFeedback, the name makes sense